tests/shap/test_log.py

```diff
@@ -0,0 +1,152 @@
+    """
+    Tests mlflow.shap log_explainer with SHAP internal serialization of the underlying model
+    """
+
+    with mlflow.start_run() as run:
+
+        run_id = run.info.run_id
+
+        X, y = shap.datasets.boston()
+        model = sklearn.ensemble.RandomForestRegressor(n_estimators=100)
+        model.fit(X, y)
+
+        explainer_original = shap.Explainer(model.predict, X, algorithm="permutation")
+        shap_values_original = explainer_original(X[:5])
+
+        mlflow.shap.log_explainer(explainer_original, "test_explainer", False)
+
+        explainer_loaded = mlflow.shap.load_explainer("runs:/" + run_id + "/test_explainer")
+        shap_values_new = explainer_loaded(X[:5])
+
+        np.testing.assert_array_equal(shap_values_original.base_values, shap_values_new.base_values)
+        np.testing.assert_allclose(
+            shap_values_original.values, shap_values_new.values, rtol=100, atol=100
+        )
+
+
+def test_sklearn_log_explainer_pyfunc():
+    """
+    Tests mlflow.shap log_explainer with mlflow
+    serialization of the underlying model using pyfunc flavor
+    """
+
+    with mlflow.start_run() as run:
+
+        run_id = run.info.run_id
+
+        X, y = shap.datasets.boston()
+        model = sklearn.ensemble.RandomForestRegressor(n_estimators=100)
+        model.fit(X, y)
+
+        explainer_original = shap.Explainer(model.predict, X, algorithm="permutation")
+        shap_values_original = explainer_original(X[:2])
+
+        mlflow.shap.log_explainer(explainer_original, "test_explainer")
+
+        explainer_pyfunc = mlflow.pyfunc.load_model("runs:/" + run_id + "/test_explainer")
+        shap_values_new = explainer_pyfunc.predict(X[:2])
+
+        np.testing.assert_allclose(shap_values_original.values, shap_values_new, rtol=100, atol=100)
+
+
+def test_load_pyfunc():
+
+    X, y = shap.datasets.boston()
+    model = sklearn.ensemble.RandomForestRegressor(n_estimators=100)
+    model.fit(X, y)
+
+    explainer_original = shap.Explainer(model.predict, X, algorithm="permutation")
+    shap_values_original = explainer_original(X[:2])
+    mlflow.shap.save_model(explainer_original, "pyfunc_test")
+
+    explainer_pyfunc = mlflow.shap._load_pyfunc("pyfunc_test")
+    shap_values_new = explainer_pyfunc.predict(X[:2])
+
+    np.testing.assert_allclose(shap_values_original.values, shap_values_new, rtol=100, atol=100)
+
+
+def test_merge_environment():
+
+    test_shap_env = {
+        "channels": ["default", "conda-forge"],
+        "dependencies": ["python=3.8.5", "pip", {"pip": ["mlflow", "shap==0.38.0"]}],
+    }
+
+    test_model_env = {
+        "channels": ["default", "conda-forge"],
+        "dependencies": [
+            "python=3.8.5",
+            "pip",
+            {"pip": ["mlflow", "scikit-learn==0.24.0", "cloudpickle==1.6.0"]},
+        ],
+    }
+
+    expected_merged_env = {
+        "name": "mlflow-env",
+        "channels": ["defaults", "conda-forge"],
```

Comment:
```suggestion
        "channels": ["default", "conda-forge"],
```

The first element should be `default`, right?

Reply:
its 'defaults' 
https://github.com/mlflow/mlflow/blob/ebd4e5ed2e12055d5b1d3ff3c740fd838301f616/mlflow/utils/environment.py#L5

Reply:
I just saw this change.
https://github.com/mlflow/mlflow/pull/3989/commits/4bbe4270fb912bda1639ac21655f81ce862fdf3e#diff-51825b2d89b3dd80ba874d5f411374ae2bb3821675a64895e9d686df84b2d542R107-R113

Makes sense to use `"defaults"` now.